bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/ProgramType.scala

```diff
@@ -7,50 +7,96 @@ sealed trait ProgramType extends Ordered[ProgramType] {
   def abbreviation: String
   override def toString = abbreviation
   def isScience: Boolean = true
+  def typeEnum: ProgramTypeEnum
 
   def compare(that : ProgramType): Int = abbreviation.compare(that.abbreviation)
 }
 
 object ProgramType {
+
   case object Calibration extends ProgramType {
     val name = "Calibration"
     val abbreviation = "CAL"
     override val isScience = false
+    val typeEnum = ProgramTypeEnum.CAL
   }
+
+  /** Access from Java code. */
+  def CAL: ProgramType = Calibration
```

Comment:
I could not figure out another way to reference `Calibration` from Java.  If somebody knows what the trick is, we can remove these.